internal/watcher/nginx_process_parser.go

```diff
@@ -0,0 +1,340 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package watcher
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
+	"github.com/nginx/agent/v3/internal/datasource/host/exec"
+	"github.com/nginx/agent/v3/internal/model"
+	"github.com/nginx/agent/v3/internal/uuid"
+)
+
+const (
+	withWithPrefix   = "with-"
+	withModuleSuffix = "module"
+)
+
+type (
+	NginxProcessParser struct {
+		executer exec.ExecInterface
+	}
+
+	Info struct {
+		ProcessID       int32
+		Version         string
+		Prefix          string
+		ConfPath        string
+		ConfigureArgs   map[string]interface{}
+		ExePath         string
+		LoadableModules []string
+		DynamicModules  []string
+	}
+)
+
+var (
+	_ processParser = (*NginxProcessParser)(nil)
+
+	versionRegex = regexp.MustCompile(`(?P<name>\S+)\/(?P<version>.*)`)
+)
+
+func NewNginxProcessParser() *NginxProcessParser {
+	return &NginxProcessParser{
+		executer: &exec.Exec{},
+	}
+}
+
+func (npp *NginxProcessParser) Parse(ctx context.Context, processes []*model.Process) []*v1.Instance {
+	processList := []*v1.Instance{}
+
+	nginxProcesses := filterNginxProcesses(processes)
+
+	for _, nginxProcess := range nginxProcesses {
+		_, ok := nginxProcesses[nginxProcess.Ppid]
+		if !ok {
```

Comment:
nit: maybe add a comment to explain what's happening here (finding the master NGINX process I believe?).